### PR TITLE
MULE-9929: Remove MuleContext from MuleMessage

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/construct/processor/PipelineMessageNotificationTestCase.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public void requestResponse() throws MuleException
         pipeline.setMessageSource(source);
         pipeline.initialise();
 
-        event = new DefaultMuleEvent(new DefaultMuleMessage("request", muleContext),
+        event = new DefaultMuleEvent(new DefaultMuleMessage("request"),
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/processor/PipelineMessageNotificationTestCase.java]: formatting

File: core/src/main/java/org/mule/runtime/core/routing/filters/logic/AndFilter.java [hunk 2]
```diff
@@ -49,4 +51,22 @@ public boolean accept(MuleMessage message)
 
         return true;
     }
+
+    @Override
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/routing/filters/logic/AndFilter.java]: How is this related to this jira issue?

Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/routing/filters/logic/AndFilter.java]: See LogicFiltersTestCase
